### ISSUE TITLE
[BUG]: Internal Server Error on /api/v1/auth/register due to MissingBackendError in passlib

### ISSUE THREAD

**Issue by jatinkrmalik** [blossomlabsio/Bloom-Backend#33]
# Description:

When attempting to register a new user via the `/api/v1/auth/register` endpoint, the server throws a `500 Internal Server Error`. The traceback points to a `MissingBackendError` from the `passlib` library, indicating that the `bcrypt` backend is not available.

# Steps to Reproduce:
1. Start the FastAPI application via `pdm run start`
2. Navigate to the `/docs` endpoint and use the UI to send a POST request to `/api/v1/auth/register`.
3. Observe the `500 Internal Server Error`.

# Expected Behavior:
The registration process should complete without errors, and the user should be successfully registered.

# Actual Behavior: 
Received a `500 Internal Server Error` due to a `MissingBackendError` in `passlib`. Logs below.

### Describe the device where you encountered the bug on

Browser: Chrome
OS: MacOS

### Relevant log output and/or screenshot(s)

```
❯ pdm run start
/Users/jmalik/github/Bloom-Backend/.venv/lib/python3.11/site-packages/pydantic/_internal/_config.py:317: UserWarning: Valid config keys have changed in V2:
* 'orm_mode' has been renamed to 'from_attributes'
  warnings.warn(message, UserWarning)
INFO:     | 2023-10-25 00:18:58 | Starting Bloom API
INFO:     Started server process [67788]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
INFO:     127.0.0.1:60551 - "GET /docs HTTP/1.1" 200 OK
INFO:     127.0.0.1:60551 - "GET /openapi.json HTTP/1.1" 200 OK
INFO:     127.0.0.1:60551 - "POST /api/v1/auth/register HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/Users/jmalik/github/Bloom-Backend/.venv/lib/python3.11/site-packages/uvicorn/protocols/http/httptools_impl.py", line 426, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jmalik/github/Bloom-Backend/.venv/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jmalik/github/Bloom-Backend/.venv/lib/python3.11/site-packages/fastapi/applications.py", line 1115, in __call__
    await super().__call__(scope, receive, send)
  File "/Users/jmalik/github/Bloom-Backend/.venv/lib/python3.11/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/Users/jmalik/github/Bloom-Backend/.venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/Users/jmalik/github/Bloom-Backend/.venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/Users/jmalik/github/Bloom-Backend/.venv/lib/python3.11/site-packages/starlette/middleware/authentication.py", line 48, in __call__
    await self.app(scope, receive, send)
  File "/Users/jmalik/github/Bloom-Backend/.venv/lib/python3.11/site-packages/starlette/middleware/cors.py", line 91, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "/Users/jmalik/github/Bloom-Backend/.venv/lib/python3.11/site-packages/starlette/middleware/cors.py", line 146, in simple_response
    await self.app(scope, receive, send)
  File "/Users/jmalik/github/Bloom-Backend/.venv/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/Users/jmalik/github/Bloom-Backend/.venv/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/Users/jmalik/.pyenv/versions/3.11.0rc2/lib/python3.11/contextlib.py", line 222, in __aexit__
    await self.gen.athrow(typ, value, traceback)
  File "/Users/jmalik/github/Bloom-Backend/.venv/lib/python3.11/site-packages/fastapi/concurrency.py", line 36, in contextmanager_in_threadpool
    raise e
passlib.exc.MissingBackendError: bcrypt: no backends available -- recommend you install one (e.g. 'pip install bcrypt')
```

**Comment by Rage-cmd** [comment]
I would like to work on it

**Comment by jatinkrmalik** [comment]
> I would like to work on it

Sorry bud, I already fixed it in #34 , just raised this issue to document the problem! :)

**Comment by Rage-cmd** [comment]
@jatinkrmalik No problem, thanks for informing me 😄
